### PR TITLE
Track leaders by slack ID

### LINKED ISSUE #22: Store Slack user ids in addition to usernames

### PR DESCRIPTION
**Pull request by paked**
Previously we've been pinging our leader's by doing a lookup of their
Slack username as recorded in our database. This has led to problems
where they then change their usernames without informing a staff member.
Subsequently, the database is now out of sync and we cannot send them
automated messages.

The Slack ID stays constant no matter what username you have, therefore
solving our problem.

Fixes #22

### CODE REVIEW

**Comment by MaxWofford** [comment]
@paked: Could you fix the tests and ping me once it's ready?

File: app/jobs/leader_check_ins_job.rb [hunk 1]
```diff
@@ -7,13 +7,12 @@ class LeaderCheckInsJob < ApplicationJob
   LEADER_PIPELINE_KEY = Rails.application.secrets.streak_leader_pipeline_key
 
   def perform(usernames = [])
-    usernames = active_leader_usernames if usernames.empty?
+    user_ids = active_leadre_slack_ids
```
Comment (MaxWofford) [inline app/jobs/leader_check_ins_job.rb]: should this be `active_leader_slack_ids`?

File: app/jobs/leader_check_ins_job.rb [hunk 2]
```diff
@@ -7,13 +7,12 @@ class LeaderCheckInsJob < ApplicationJob
...
 
-    usernames.each do |username|
-      user = user_from_username(username)
-      next if user.nil? # couldn't find the user
+    user_ids = user_ids_from_usernames usernames unless usernames.empty
```
Comment (MaxWofford) [inline app/jobs/leader_check_ins_job.rb]: Isn't this redundant after https://github.com/hackclub/api/pull/94/files#diff-901c18c30d3bc30f13b107f181a0f6fdR10?

Comment (paked) [inline app/jobs/leader_check_ins_job.rb]: No, this statement checks whether it has been provided a list of usernames – if it has it only starts check ins with those, as opposed to every active leader.

File: app/jobs/leader_check_ins_job.rb (hunk 1 above)
Comment (paked) [inline app/jobs/leader_check_ins_job.rb]: Yes.

**Comment by paked** [comment]
@MaxWofford: Addressed.

**Pull request merged**